### PR TITLE
Docs: CRD reference topic edits

### LINKED ISSUE #2451: Docs: Edit and  build Agent Operator CRD reference docs
Edited 

agent/operator/apis/monitoring/v1alpha1/
   deployment.go
   doc.go
   group.go
   types_integrations.go
   types_logs.go

Will finish editing topics in this issue: https://github.com/grafana/agent/issues/2452

### PR DESCRIPTION
**Pull request by karengermond**
#### PR Description

Edited

agent/operator/apis/monitoring/v1alpha1/
deployment.go
doc.go
group.go
types_integrations.go
types_logs.go

Will finish editing topics in this issue: https://github.com/grafana/agent/issues/2452

#### Which issue(s) this PR fixes

Fixes #2451

#### PR Checklist

- [ ] CHANGELOG updated
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: cmd/agent-operator/DEVELOPERS.md [hunk 1]
```diff
@@ -41,6 +41,12 @@ go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.6.2
 go install github.com/ahmetb/gen-crd-api-reference-docs@v0.3.1-0.20220618162802-424739b250f5
 ```
 
+Use the following to run the script in a container:
+
+```
+docker build -t grafana/agent-build-image:0.18.0 ./build-image
```
Comment (marctc) [inline cmd/agent-operator/DEVELOPERS.md:47]: This version number is too specific, maybe we could just use `latests` as a tag.

**Review by marctc — APPROVED** [review]
Thanks! Left one minor comment, but the rest LGTM

File: cmd/agent-operator/DEVELOPERS.md (hunk 1 above)
Comment (karengermond) [inline cmd/agent-operator/DEVELOPERS.md:47]:
```suggestion
docker build -t grafana/agent-build-image:latest ./build-image
```

File: cmd/agent-operator/DEVELOPERS.md [hunk 2]
```diff
@@ -41,6 +41,12 @@ go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.6.2
...
+BUILD_IMAGE=grafana/agent-build-image:0.18.0 USE_CONTAINER=1 make generate-crds
```
Comment (karengermond) [inline cmd/agent-operator/DEVELOPERS.md:48]:
```suggestion
BUILD_IMAGE=grafana/agent-build-image:latest USE_CONTAINER=1 make generate-crds
```

**Pull request merged**